hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/utils/Pipelines.java

```diff
@@ -410,10 +410,11 @@ public static DataStream<Object> hoodieStreamWrite(Configuration conf, DataStrea
    * @return the compaction pipeline
    */
   public static DataStreamSink<CompactionCommitEvent> compact(Configuration conf, DataStream<Object> dataStream) {
-    return dataStream.transform("compact_plan_generate",
+    DataStreamSink<CompactionCommitEvent> compactionCommitEventDataStream = dataStream.transform("compact_plan_generate",
             TypeInformation.of(CompactionPlanEvent.class),
             new CompactionPlanOperator(conf))
         .setParallelism(1) // plan generate must be singleton
+        .setMaxParallelism(1)
```

Comment:
Is this line compatible with flink release before 1.18?

Reply:
compatible.

Only the DataStreamSink Class doesn't  have  setMaxParallelism method before 1.18.
https://issues.apache.org/jira/browse/FLINK-31873

Reply:
The SingleOutputStreamOperator Class was returned when call  Datastream.transform  method. 
SingleOutputStreamOperator class  contains the setMaxParallelism method.

For example :  flink 1.14
https://github.com/apache/flink/blob/08dbb6d5eff46e72af412643f80ac353636821de/flink-streaming-java/src/main/java/org/apache/flink/streaming/api/datastream/SingleOutputStreamOperator.java#L155